configs/rails_config.yml

```diff
@@ -110,27 +116,35 @@ jobs:
       - run:
           name: "Compile and serve assets"
           command: |
-            if [ $CIRCLE_BRANCH == staging ]
-            then
-              export ASSETS_ENV=staging
-              export S3_ASSETS_BUCKET=<< pipeline.parameters.assets-bucket-staging >>
-            elif [ $CIRCLE_BRANCH == master ]
-            then
-              export ASSETS_ENV=production
-              export S3_ASSETS_BUCKET=<< pipeline.parameters.assets-bucket-production >>
-            else
-              exit 0
+            branches=<< pipeline.parameters.branches-for-assets >>
+            assetsBuckets=<< pipeline.parameters.assets-buckets >>
+            if [ ${#assetsBuckets[@]} -eq 0 ]; then
+              assetsBuckets=('<< pipeline.parameters.assets-bucket-production >>' '<< pipeline.parameters.assets-bucket-staging >>')
             fi
 
-            yarn
-            RAILS_ENV=${ASSETS_ENV} bundle exec rake db:create db:migrate
-            RAILS_ENV=${ASSETS_ENV} bundle exec rake assets:clobber assets:precompile
-            aws s3 cp public/assets/ s3://${S3_ASSETS_BUCKET}/assets/ --recursive --acl public-read
+            for i in "${!branches[@]}"; do
```

Comment:
Le `!` ici est intentionnel ou c'est une typo ? 😕 Si c'est intentionnel je ne comprends pas pourquoi il est là.

Si `$branches` est égal à `('staging' 'master:production' 'other-branch')` alors le `!` va faire en sorte que `'staging'`, `'master:production'` et `'other-branch'  soient utilisés comme le nom des variables, et ces variables n'existent pas, et donc ça donnera un truc du genre `for i in "" "" ""; do`.

Reply:
En fait je pense que je me trompe, d'après le [manual de BASH](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html):

```
${!name[@]}
${!name[*]}

    If name is an array variable, expands to the list of array indices (keys) assigned in name. If name is not an array, expands to 0 if name is set and null otherwise. When ‘@’ is used and the expansion appears within double quotes, each key expands to a separate word.
```

Mais du coup ma question c'est:

Si tu as utilisé `!` c'est que tu l'as testé et que ça marche, mais du coup c'est quoi la difference avec `for i in "${branches[@]}"` ?

Reply:
> `$i` correspond a l'index de cette branche dans ta liste (c'est le `i` de `for i in "${!branches[@]}"`).

Ok donc tu loop bien volontairement sur les index. 🤔